src/twisted/internet/endpoints.py

```diff
@@ -716,6 +809,30 @@ def __init__(self, reactor, host, port, timeout=30, bindAddress=None,
         self._attemptDelay = attemptDelay
 
 
+    def _getNameResolverAndMaybeWarn(self, reactor):
+        """
+        Retrieve a C{nameResolver} callable and warn the caller's
+        caller that using a reactor which doesn't provide
+        L{IReactorPluggableNameResolver} is deprecated.
+
+        @param reactor: The reactor to check.
+
+        @return: A L{IHostnameResolver} provider.
+        """
+        if not IReactorPluggableNameResolver.providedBy(reactor):
+            warningString = deprecate.getDeprecationWarningString(
+                reactor.__class__,
+                Version("Twisted", "NEXT", 0, 0),
+                format=("Passing HostnameEndpoint %(fqpn)s"
+                        " was deprecated in %(version)s"),
```

Comment:
Including the FQPN of the reactor seems fine. But I'd go with something like:

> Passing a reactor that doesn't implement IReactorPluggableNameResolver to HostnameEndpoint was deprecated <...>. (`%(fqpn)s` was passed, which doesn't currently implement <...>

that makes it clear that a possible fix is making the reactor implement the given interface.